### PR TITLE
Remove references to ioutil

### PR DESCRIPTION
**Pull request by barkbay**
`io/ioutil` is deprecated since Go 1.16 and reported as such when `golangci-lint` is upgraded to its latest version.

### CODE REVIEW

File: hack/deployer/runner/kind.go [hunk 1]
```diff
@@ -97,7 +96,7 @@ func (k *KindDriver) create() error {
 	if err != nil {
 		return err
 	}
-	defer os.Remove(tmpManifest.Name())
+	defer os.RemoveAll(tmpManifest.Name())
```
Comment (thbkrkr) [inline hack/deployer/runner/kind.go:99]: Why this change? Isn't `tmpManifest` a file?

File: pkg/utils/test/webhook.go [hunk 2]
```diff
@@ -77,6 +76,7 @@ func ValidationWebhookFailed(causeRegexes ...string) func(*testing.T, *admission
 }
 
 // RunValidationWebhookTests runs a series of ValidationWebhookTestCases
+//
```
Comment (thbkrkr) [inline pkg/utils/test/webhook.go:79]: Do we need this?

**Review by thbkrkr — APPROVED** [review]
LGTM

File: hack/deployer/runner/kind.go (hunk 1 above)
Comment (barkbay) [reply hack/deployer/runner/kind.go:99]: I used a script to generate this PR, something may not have worked as expected here 😬  I'll fix it.

File: pkg/utils/test/webhook.go (hunk 2 above)
Comment (barkbay) [reply pkg/utils/test/webhook.go:79]: I'm confused, this line is automatically added by my IDE, I'm trying to understand why 🤔

Comment (barkbay) [reply pkg/utils/test/webhook.go:79]: This is actually added by `gofmt` 🤷

**Comment by barkbay** [comment]
jenkins test this please

File: pkg/utils/test/webhook.go (hunk 2 above)
Comment (barkbay) [reply pkg/utils/test/webhook.go:79]: > This is actually added by gofmt 🤷

Just realised that it is because I'm using [Go 1.19](https://tip.golang.org/doc/go1.19#go-doc):

> Go 1.19 adds support for links, lists, and clearer headings in doc comments. As part of this change, [gofmt](https://tip.golang.org/cmd/gofmt) now reformats doc comments to make their rendered meaning clearer.
